### PR TITLE
General charm/pet fixes

### PR DESCRIPTION
**Pull request by cyberium**
- Now creatures can be charmed.
- Now players can be charmed.
- We can now charm/possess even if we have already a summoned pet.
- Charmed pet have their own threatlist. But their original one before they were charmed is preserved.
- Charmer have now attack spell in pet bar even if he charmed a player.
- Its now possible to charm a player while dueling.
- Charmed players/creature will not be teleported if areatrigger is triggered.
- Uncharm before telport.
- We can now stable unsumoned pet.
- If we are in flight while loading the pet will be set as temporary unsumoned and will be correctly resummoned at landing.
- Priest pet is now not saved to db.
- Its now not possible to summon/stable a dead pet. If its not summoned we are now able to revive it from DB.
- Client error message are improved in different case like trying to summon a dead pet.
- Now in combat state is applied only after spell hit. (before it was impossible to cast a spell with not in combat attribute)

### CODE REVIEW

**Comment by Jose123456** [comment]
Works fine for me. I haven't had a chance to test any of the pet fixes. I'll try to this weekend.

File: src/game/Pet.cpp [hunk 1]
```diff
@@ -89,7 +87,73 @@ void Pet::RemoveFromWorld()
...
+        // current pet (slot 0)
+        request << "FROM character_pet WHERE owner = '" << ownerid << "' AND slot = '" << uint32(PET_SAVE_AS_CURRENT) << "'";
+    else if (petentry)
+        // known petentry entry (unique for summoned pet, but non unique for hunter pet (only from current or not stabled pets)
+        request << "FROM character_pet WHERE owner = '" << ownerid << "' AND entry = '" << petentry << "' AND (slot = '" << uint32(PET_SAVE_AS_CURRENT) << "' OR slot > '" << uint32(PET_SAVE_LAST_STABLE_SLOT) << "')";
+    else
+        // any current or other non-stabled pet (for hunter "call pet")
+        request << "FROM character_pet WHERE owner = '" << ownerid << "' AND (slot = '" << uint32(PET_SAVE_AS_CURRENT) << "' OR slot > '" << uint32(PET_SAVE_LAST_STABLE_SLOT) << "')";
+
+    QueryResult* result = CharacterDatabase.PQuery(request.str().c_str());
```
Comment (Jose123456) [inline src/game/Pet.cpp]: This line generates a compiler warning in Linux:
```
mangos-classic/src/game/Pet.cpp: In static member function ‘static SpellCastResult Pet::TryLoadFromDB(Unit*, uint32, uint32, bool, PetType)’:
mangos-classic/src/game/Pet.cpp:111:73: warning: format not a string literal and no format arguments [-Wformat-security]
     QueryResult* result = CharacterDatabase.PQuery(request.str().c_str());
                                                                         ^
```

**Comment by cyberium** [comment]
merged thanks to all

**Comment by HanFox** [comment]
@cyberium since this PR was merged I'm getting a crash:
`./mangosd: symbol lookup error: ../lib/libmangosscript.so: undefined symbol: _ZNK4Unit22isInAccessablePlaceForEPK8Creature`

That look like anything to you?

**Pull request closed without merging**